modin/backends/pandas/query_compiler.py

```diff
@@ -1828,6 +1832,56 @@ def applyier(df, internal_indices, other=[], internal_other_indices=[]):
     # END Map across rows/columns
 
     # __getitem__ methods
+    def getitem_array(self, key):
```

Comment:
Rename to `filter_rows`? This name follows pandas too much I would prefer something slightly more generic I think. Do you have a better suggestion?

Reply:
`filter_rows` would be inaccurate because it can be used for both rows and columns selection.

Reply:
Yes, that makes sense.